tools/starman-development.psgi

```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/plackup 
+
+package LedgerSMB::FCGI;
+
+use CGI::Emulate::PSGI;
+use LedgerSMB::PSGI;
+use Plack::Builder;
+use Plack::Middleware::Static;
+
+
+my $app = LedgerSMB::PSGI::app();
+
+builder {
+    enable "Plack::Middleware::Static",
+       path => qr{(^/?(images|doc|UI|css)/|favicon\.ico)}, root => '.';
+
+    enable 'Debug',  panels => [
+            qw(Environment Response Timer Memory Session),
+            [ 'DBITrace', level => 2 ],
+            [ 'Profiler::NYTProf', exclude => [qw(.*\.css .*\.png .*\.ico .*\.js .*\.gif)], minimal => 1 ],
+    ] if $ENV{PLACK_ENV} =~ "development";
+
+#    # Enable Stacktrace when being accessed from the local network
+#    enable_if { $_[0]->{REMOTE_ADDR} =~ /^192\.168\.30\.*/ } 'StackTrace';
+
+#    enable 'StackTrace', unsafe_ref_capture => 1;
+    enable 'InteractiveDebugger'; # if $ENV{PLACK_ENV} =~ "development";
+    enable "HTTPExceptions", rethrow => 1; # if $ENV{PLACK_ENV} =~ "development";
```

Comment:
Yes we will have to, as soon as we decide on the proper setup. At this
point, this is still a proposal.
The profiler takes a bit of time, we may want to make it optional with a
switch.

Le 2016-04-22 15:04, Erik Huelsmann a écrit :
>
> In tools/starman-development.psgi
> <https://github.com/ehuelsmann/LedgerSMB/pull/240#discussion_r60788012>:
>
> > +builder {
> > +    enable "Plack::Middleware::Static",
> > +       path => qr{(^/?(images|doc|UI|css)/|favicon\.ico)}, root => '.';
> > +
> > +    enable 'Debug',  panels => [
> > +            qw(Environment Response Timer Memory Session),
> > +            [ 'DBITrace', level => 2 ],
> > +            [ 'Profiler::NYTProf', exclude => [qw(.*\.css .*\.png .*\.ico .*\.js .*\.gif)], minimal => 1 ],
> > +    ] if $ENV{PLACK_ENV} =~ "development";
> > +
> > +#    # Enable Stacktrace when being accessed from the local network
> > +#    enable_if { $_[0]->{REMOTE_ADDR} =~ /^192\.168\.30\.*/ } 'StackTrace';
> > +
> > +#    enable 'StackTrace', unsafe_ref_capture => 1;
> > +    enable 'InteractiveDebugger'; # if $ENV{PLACK_ENV} =~ "development";
> > +    enable "HTTPExceptions", rethrow => 1; # if $ENV{PLACK_ENV} =~ "development";
>
> Should we add these (debug, interactivedebugger and httpexceptions) as
> Development requirements in Makefile.PL?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/ehuelsmann/LedgerSMB/pull/240/files/a6636f6dc8d0aad08544168b52862b70c41ba586#r60788012>
>